extensions/amp-story/0.1/bookend/bookend-component.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ArticleComponent, ArticleTitle} from './components/article';
+
+/**
+ * @typedef {{
+ *   bookend-vesion: string,
+ *   components: !Array<!BookendComponentDef>,
+ * }}
+ */
+export let BookendDataDef;
+
+/**
+ * @typedef {{
+ *   type: string,
+ *   share-providers: (Array<*>|undefined),
+ *   title: (string|undefinded),
+ *   url: (string|undefinded),
+ *   image: (string|undefinded)
+ * }}
+ */
+export let BookendComponentDef;
```

Comment:
I think we should explicitly typedef the different components.  Like:

```javascript
/**
 * @typedef {{
 *   type: string,
 *   title: string,
 *   url: string,
 *   image: (string|undefined)
 * }}
 */
let BookendArticleComponentDef;
```


```javascript
/**
 * @typedef {{
 *   type: string,
 *   text: !Array<string>
 * }}
 */
let BookendTextComponentDef;
```

Then, union the different components for this:

```javascript
/** @typedef {!BookendArticleComponentDef|!BookendTextComponentDef|...} */
let BookendComponentDef;
```